cmd/collector/main.go

```diff
@@ -262,3 +276,37 @@ func checkForCollectorRollbackConfig(configPath string) error {
 
 	return nil
 }
+
+func configureTLS() (*opamp.TLSConfig, error) {
+	tlsConfig := opamp.TLSConfig{}
+	tlsConfigured := false
+
+	if skipVerify := os.Getenv(tlsSkipVerifyENV); skipVerify != "" {
+		s, err := strconv.ParseBool(skipVerify)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value '%s' for environment option '%s': %w", skipVerify, tlsSkipVerifyENV, err)
+		}
+		tlsConfig.InsecureSkipVerify = s
+		tlsConfigured = true
+	}
+
+	if ca := os.Getenv(tlsCaENV); ca != "" {
+		tlsConfig.CAFile = &ca
+		tlsConfigured = true
+	}
+
+	if crt := os.Getenv(tlsCertENV); crt != "" {
+		tlsConfig.CertFile = &crt
+		tlsConfigured = true
+	}
+
+	if key := os.Getenv(tlsKeyENV); key != "" {
+		tlsConfig.KeyFile = &key
+		tlsConfigured = true
+	}
+
+	if tlsConfigured {
+		return &tlsConfig, nil
+	}
+	return nil, nil
```

Comment:
I'm wondering what happens if we give opamp and empty TLSConfig? Would it correctly not do TLS? If so it would be nice to just `return tlsConfig, nil` always.

Reply:
Ah ok I forgot it was being saved back. I can't think of a cleaner way to do this so looks good.